### ISSUE TITLE
Help to add a mount to the repo example

### ISSUE THREAD

**Issue by eddieajau** [panva/node-oidc-provider#153]
Hi and apologies in advance for a newbie question.

I'm trying to modify [the example](https://github.com/panva/node-oidc-provider/tree/master/example) to mount the routes per the instructions in [Mounting oidc-provider](https://github.com/panva/node-oidc-provider/blob/master/docs/configuration.md#mounting-oidc-provider).

I've gone back through the history where this was previously included but it seems things have changed a bit since then. I'm new to Koa so I was wondering if you could put my on the right track on how to add the mounting recommendation to the example code (the example is otherwise pretty much exactly what I need to my project)?

Thanks in advance.

**Comment by panva** [comment]
I'm afraid i don't follow :) You either have an existing koa or express app and then you just follow what's in the configuration.md you linked ...

**Comment by eddieajau** [comment]
Hi @panva. So I have taken, verbatim, the example from `node-oidc-provider/example/` in this repo. I'm simply running `node index.js`. Other than modifying the `Account` code, and few tweaks here and there, it's all I need.

I don't have an existing app yet because all I want to do is stand up a very simple ODIC server as a standalone microservice in front of a bespoke business system so that I can plug it into Red Hat's Keycloak for authentication. At work we just want to use this technique whenever a client is using some crazy home-grown system, or a system that doesn't come any sort of modern authentication system (or even SAML).

The microservice will be running as a Docker container so it's a lot easier for the Dev Ops guy to mount a sub-path with his load balancer thingy rather than me giving him a list of all different routes that oidc-provider mounts at the root level.

In other words, `provider.app` IS the only app running in this microservice.

Is that a little clearer? Thanks in advance.

**Comment by eddieajau** [comment]
Hrm, can I just inject this into the config:

```
  routes: {
    authorization: '/oidc/auth',
    certificates: '/oidc/certs',
    check_session: '/oidc/session/check',
    end_session: '/oidc/session/end',
    introspection: '/oidc/token/introspection',
    registration: '/oidc/reg',
    revocation: '/oidc/token/revocation',
    token: '/oidc/token',
    userinfo: '/oidc/me',
  },
```

**Comment by panva** [comment]
Yes. https://github.com/panva/node-oidc-provider/blob/master/docs/configuration.md#configuring-routes

**Comment by eddieajau** [comment]
Thank you.

**Comment by jonenst** [comment]
Hi @panva , can you confirm that when using the standalone mode, configuring all the routes with the prefix is still the only solution to host at a path like "/oidc" ?
Do you have any recommendation to avoid the verbosity ? Is it recommended to not use the standalone version, and instead use one of the the other methods ? Which one is most recommended instead of the standalone version for simplicity ?

Thanks in advance,
Jon

**Comment by panva** [comment]
https://github.com/panva/node-oidc-provider/blob/v6.23.3/docs/README.md#mounting-oidc-provider

**Comment by jonenst** [comment]
Hi, thanks for the quick answer. I had seen that section, and didn't find the standalone version of it. Is it not recommended ?

**Comment by panva** [comment]
Hi Jon, your point was simplicity, in that case - just mount it to your existing app which you'll most likely use to define interactions anyway. Hm?

As for verbosity? I don't get the point.

**Comment by jonenst** [comment]
Thanks again for replying.
From the user browser perspective, I want the server to respond to
```
XXX/oidc/.well-known/openid-configuration
```
The returned json should contain urls to like
```
XXX/oidc-server/auth
```

Is it possible to achieve that with only the node-oidc-provider server in standalone mode ? Is it possible and better with a node-oidc-provider server using another mode ? (express, koa, whatever)

**Comment by panva** [comment]
Well that's definitely not possible with standalone because discovery is mounted to root and you have no way of changing that. So you gotta mount, and mount to whatever you feel like you can work with best.

**Comment by jonenst** [comment]
Thank you, I understand now.

Maybe you should add in the mouting section that you referenced that mounting is not possible with the standalone version, even if it sounds obvious to you. I guess the best option is to use the koa mount since that's what you use internally.
